File: pkg/linters/python/testdata/valid.py
"""A valid Python file with proper formatting and no issues."""


def greet(name: str) -> str:
    """Return a greeting message."""
    return f"Hello, {name}!"


def add(a: int, b: int) -> int:
    """Add two numbers together."""
    return a + b


class Calculator:
    """A simple calculator class."""

    def __init__(self) -> None:
        """Initialize the calculator."""
        self.result = 0

    def add(self, value: int) -> None:
        """Add a value to the result."""
        self.result += value

    def get_result(self) -> int:
        """Get the current result."""
        return self.result


def main() -> None:
    """Main function."""
    print(greet("World"))
    print(f"2 + 3 = {add(2, 3)}")

    calc = Calculator()
    calc.add(10)
    calc.add(20)
    print(f"Calculator result: {calc.get_result()}")


if __name__ == "__main__":
    main()

File: pkg/linters/rust/testdata/valid.rs
// A valid Rust file with proper formatting and no issues

fn main() {
    println!("Hello, world!");
    
    let numbers = vec![1, 2, 3, 4, 5];
    let sum: i32 = numbers.iter().sum();
    
    println!("Sum: {}", sum);
    
    if sum > 10 {
        println!("Sum is greater than 10");
    } else {
        println!("Sum is less than or equal to 10");
    }
}

#[derive(Debug)]
struct Person {
    name: String,
    age: u32,
}

impl Person {
    fn new(name: &str, age: u32) -> Self {
        Person {
            name: name.to_string(),
            age,
        }
    }
    
    fn greet(&self) {
        println!("Hello, my name is {} and I'm {} years old", self.name, self.age);
    }
}

#[cfg(test)]
mod tests {
    use super::*;
    
    #[test]
    fn test_person_creation() {
        let person = Person::new("Alice", 30);
        assert_eq!(person.name, "Alice");
        assert_eq!(person.age, 30);
    }
}